sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -257,3 +258,28 @@ object PhysicalAggregation {
     case _ => None
   }
 }
+
+/**
+ * A pattern that matches a base table access and collects the
+ * the corresponding leaf node and the applied predicates.
+ * A base table access represents a LeafNode, or a Project or Filter operators
+ * above a LeafNode.
+ */
+object BaseTableAccess extends PredicateHelper {
+  def unapply(plan: LogicalPlan): Option[(LogicalPlan, Seq[Expression])] = plan match {
```

Comment:
Is this the same a `PhysicalOperation`? Or, better question, what is the difference?